### ISSUE TITLE
random failing ServiceSupplierTestCase.testOptionalReferences

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform#488]
on win
```
java.lang.AssertionError: expected:<1> but was:<2>
	at org.eclipse.e4.core.internal.tests.di.extensions.ServiceSupplierTestCase.testOptionalReferences(ServiceSupplierTestCase.java:267)
```

**Comment by HeikoKlare** [comment]
Example for fail: https://github.com/eclipse-platform/eclipse.platform/actions/runs/5342777773/jobs/9685299025

**Comment by HeikoKlare** [comment]
Test failed again with a different error message in recent build
https://github.com/eclipse-platform/eclipse.platform/pull/595/checks?check_run_id=15421937193

```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertNotNull(Assert.java:713)
	at org.junit.Assert.assertNotNull(Assert.java:723)
	at org.eclipse.e4.core.internal.tests.di.extensions.ServiceSupplierTestCase.testOptionalReferences(ServiceSupplierTestCase.java:254)
```

**Comment by Michael5601** [comment]
Failed again in this [run](https://github.com/eclipse-platform/eclipse.platform/pull/605/checks?check_run_id=15704942223) in PR: https://github.com/eclipse-platform/eclipse.platform/pull/605

**Comment by fedejeanne** [comment]
Failed again in [this build](https://github.com/eclipse-platform/eclipse.platform/actions/runs/7030446647/job/19129999755?pr=776) of #776 

```java
org.eclipse.e4.core.internal.tests.di.extensions.ServiceSupplierTestCase.testOptionalReferences -- Time elapsed: 0.070 s <<< FAILURE!
java.lang.AssertionError: expected same:<class org.eclipse.e4.core.internal.tests.di.extensions.DisabledServiceA> was not:<class org.eclipse.e4.core.internal.tests.di.extensions.DisabledServiceB>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotSame(Assert.java:829)
	at org.junit.Assert.assertSame(Assert.java:772)
	at org.junit.Assert.assertSame(Assert.java:783)
	at org.eclipse.e4.core.internal.tests.di.extensions.ServiceSupplierTestCase.testOptionalReferences(ServiceSupplierTestCase.java:270)
```

**Comment by HeikoKlare** [comment]
It is a different kind of failure after #829. I'll have a look.

**Comment by jukzi** [comment]
i can not reproduce it locally

**Comment by jukzi** [comment]
please change the classname of the new test to avoid confusion

**Comment by jukzi** [comment]
in stacktraces the package name is visible, while in some junit results it's not obvious

**Comment by HeikoKlare** [comment]
I see. One thing that can currently help is to look at the log and see which project was processed (which may also not be easy due to parallel build).

Concerning this specific issue the log shows that the error happens in the javax copy of the tests, which is the one that has not been corrected in #829. I will provide a PR to fix that.
